app/questionnaire/path_finder.py

```diff
@@ -176,6 +166,39 @@ def _evaluate_routing_rules(
 
                 return next_block_index
 
+    def _evaluate_skip_conditions(
+        self, this_location, routing_path_block_ids, skip_conditions
+    ):
+        if not skip_conditions:
+            return False
+
+        when_rule_evaluator = WhenRuleEvaluator(
+            self.schema,
+            self.answer_store,
+            self.list_store,
+            self.metadata,
+            self.response_metadata,
+            location=this_location,
+            routing_path_block_ids=routing_path_block_ids,
+        )
+
+        if isinstance(skip_conditions, dict):
+            return when_rule_evaluator.evaluate(skip_conditions["when"])
```

Comment:
Can be moved down to only initialise when needed.
```suggestion
        if isinstance(skip_conditions, dict):
            when_rule_evaluator = WhenRuleEvaluator(
                self.schema,
                self.answer_store,
                self.list_store,
                self.metadata,
                self.response_metadata,
                location=this_location,
                routing_path_block_ids=routing_path_block_ids,
            )

            return when_rule_evaluator.evaluate(skip_conditions["when"])
```